driver-core/src/test/functional/com/mongodb/client/model/expressions/StringExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,156 @@
+
+    private final String sushi = "\u5BFF\u53F8";
+    private final String jalapeno = "jalape\u00F1o";
+
+    @Test
+    public void literalsTest() {
+        assertExpression("", of(""), "''");
+        assertExpression("abc", of("abc"), "'abc'");
+        assertThrows(IllegalArgumentException.class, () -> of((String) null));
+        assertExpression(sushi, of(sushi), "'" + sushi + "'");
+    }
+
+    @Test
+    public void concatTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/concat/
+        assertExpression(
+                "abc".concat("de"),
+                of("abc").concat(of("de")),
+                "{'$concat': ['abc', 'de']}");
+    }
+
+    @Test
+    public void toLowerTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/toLower/
+        assertExpression(
+                "ABC".toLowerCase(),
+                of("ABC").toLower(),
+                "{'$toLower': 'ABC'}");
+    }
+
+    @Test
+    public void toUpperTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/toUpper/ (?)
+        assertExpression(
+                "abc".toUpperCase(),
+                of("abc").toUpper(),
+                "{'$toUpper': 'abc'}");
+    }
+
+    @Test
+    public void strLenTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/strLenCP/ (?)
+        // TODO naming: just strLen, lengthCP, lengthCodePoints, stringLengthInCodePoints?...
+        assertExpression(
+                "abc".length(),
+                of("abc").strLen(),
+                "{'$strLenCP': 'abc'}");
+
+        // unicode
+        assertExpression(
+                jalapeno.length(),
+                of(jalapeno).strLen(),
+                "{'$strLenCP': '" + jalapeno + "'}");
+        assertExpression(
+                sushi.length(),
+                of(sushi).strLen(),
+                "{'$strLenCP': '" + sushi + "'}");
+    }
+
+    @Test
+    public void strLenBytesTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/strLenBytes/ (?)
+        assertExpression(
+                "abc".getBytes(StandardCharsets.UTF_8).length,
+                of("abc").strLenBytes(),
+                "{'$strLenBytes': 'abc'}");
+
+        // unicode
+        assertExpression(
+                jalapeno.getBytes(StandardCharsets.UTF_8).length,
+                of(jalapeno).strLenBytes(),
+                "{'$strLenBytes': '" + jalapeno + "'}");
+        assertExpression(
+                sushi.getBytes(StandardCharsets.UTF_8).length,
+                of(sushi).strLenBytes(),
+                "{'$strLenBytes': '" + sushi + "'}");
+
+        // comparison
+        assertExpression(2, of(sushi).strLen());
+        assertExpression(6, of(sushi).strLenBytes());
+        assertExpression(8, of(jalapeno).strLen());
+        assertExpression(9, of(jalapeno).strLenBytes());
```

Comment:
The assertions with magic numbers (2, 6, ...) check nothing in addition to what the assertions above check, but are confusing to a reader. Let's remove them.

Reply:
If a human can't write the expected value on his own, which is the case in these tests, then the expected value is just magic and the corresponding assertion is almost of no use. This is especially true when you have assertions that do the same but without using magic numbers.

> They demonstrate the actual expected values

What is the use of this information?

Reply:
Discussed in Zoom.